src/components/App/index.js

```diff
@@ -0,0 +1,111 @@
+import { h,  Component } from 'preact'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import EventEmitter from 'eventemitter2'
+import Modal from '../Modal'
+import Router from '../Router'
+import * as Tracker from '../../Tracker'
+import ReduxAppWrapper from '../ReduxAppWrapper/'
+import { LocaleProvider } from '../../locales'
+import { enabledDocuments } from '../Router/StepComponentMap'
+import { actions } from '../ReduxAppWrapper/store/actions/'
+import { valideJwT } from '~utils/jwt'
+
+class ModalApp extends Component {
+  constructor(props) {
+    super(props)
+    this.events = new EventEmitter()
+    this.events.on('complete', this.trackOnComplete)
+    Tracker.setUp()
+    Tracker.install()
+    this.bindEvents(props.options)
+  }
+
+  componentDidMount() {
+    this.prepareInitialStore(this.props.options)
+  }
+
+  componentDidUpdate(prevProps) {
+    this.jwtValidation(prevProps.options, this.props.options)
+    this.prepareInitialStore(this.props.options, prevProps.options)
+    this.rebindEvents(prevProps.options, this.props.options);
+  }
+
+  componentWillUnmount() {
+    this.props.socket && this.props.socket.close()
+    this.events.removeAllListeners('complete', 'error')
+    Tracker.uninstall()
+  }
+
+  jwtValidation = (prevOptions = {}, newOptions = {}) => {
+    if (prevOptions.token !== newOptions.token) {
+      if ( !valideJwT(newOptions.token) ){
+        this.onInvalidJWT()
+      }
+    }
+  }
+  
+  onInvalidJWT = () => {
+    const type = 'exception'
+    const message = 'Invalid token'
+    this.events.emit('error', { type, message })
+  }
+  
+  trackOnComplete = () => Tracker.sendEvent('completed flow')
+
+  bindEvents = ({onComplete, onError}) => {
+    this.events.on('complete', onComplete)
+    this.events.on('error', onError)
+  }
+  
+  rebindEvents = (oldOptions, newOptions) => {
+    this.events.off('complete', oldOptions.onComplete)
+    this.events.off('error', oldOptions.onError)
+    this.bindEvents(newOptions)
```

Comment:
Why not just pass in only the events props needed rather than the entire `newOptions` object?